bundles/org.openhab.binding.cbus/src/main/java/org/openhab/binding/cbus/handler/CBusCGateHandler.java

```diff
@@ -0,0 +1,433 @@
+            if (cGateSession.getSessionID() != null
+                    && contents[1].contains("sessionId=" + cGateSession.getSessionID())) {
+                // We created this event - don't worry about processing it again...
+                return;
+            }
+            // By Marking this as Nullable it fools the static analyser into understanding that poll can return Null
+            LinkedList<@Nullable String> tokenizer = new LinkedList<>(Arrays.asList(contents[0].split("\\s+")));
+            @Nullable
+            String firstToken = tokenizer.poll();
+            if (firstToken == null) {
+                logger.debug("ProcessStateChange: Cant tokenize status {}", status);
+                return;
+            }
+            switch (firstToken) {
+                case "lighting": {
+                    @Nullable
+                    String state = tokenizer.poll();
+                    @Nullable
+                    String address = tokenizer.poll();
+                    if ("ramp".equals(state)) {
+                        state = tokenizer.poll();
+                    }
+                    updateGroup(address, state);
+                    break;
+                }
+                case "temperature": {
+                    // For temperature we ignore the state
+                    tokenizer.poll();
+                    @Nullable
+                    String address = tokenizer.poll();
+                    @Nullable
+                    String temp = tokenizer.poll();
+                    updateGroup(address, temp);
+                    break;
+                }
+                case "trigger": {
+                    @Nullable
+                    String command = tokenizer.poll();
+                    @Nullable
+                    String address = tokenizer.poll();
+                    if ("event".equals(command)) {
+                        @Nullable
+                        String level = tokenizer.poll();
+                        updateGroup(address, level);
+                    } else if ("indicatorkill".equals(command)) {
+                        updateGroup(address, "-1");
+                    } else {
+                        logger.warn("Unhandled trigger command {} - status {}", command, status);
+                    }
+                    break;
+                }
+                case "clock": {
+                    @Nullable
+                    String address = "";
+                    @Nullable
+                    String value = "";
+                    @Nullable
+                    String type = tokenizer.poll();
+                    if ("date".equals(type)) {
+                        address = tokenizer.poll() + "/1";
+                        value = tokenizer.poll();
+                    } else if ("time".equals(type)) {
+                        address = tokenizer.poll() + "/0";
+                        value = tokenizer.poll();
+                    } else if (!"request_refresh".equals(type)) {
```

Comment:
You don't handle `request_refresh`?

Reply:
Added a comment about why we dont handle it.